### ISSUE TITLE
missing system calls in linux

### ISSUE THREAD

**Issue by m000** [panda-re/panda#375]
I was playing with the new `on_all_sys_enter2` callback of `syscalls2` when I experienced an unexpected crash. Investigating, I found that a syscall with number `0x163` (355 decimal) was executing which set the index out of the syscall information array.

The `0x163` syscall is not included in `syscalls2/prototypes/linux_x86_prototypes.txt` and apparently it has been added in fairly recently. How is `linux_x86_prototypes.txt` extracted so I add the syscalls missing for my kernel?

My guest OS is Ubuntu 16.04 with the 4.4.0-130-generic kernel.

**Comment by moyix** [comment]
It was originally created with this script: https://github.com/moyix/panda/blob/master/qemu/panda_plugins/syscalls2/createLinuxPrototypes.py

But it looks a bit hacky so you may not want to use it...

**Comment by m000** [comment]
This can probably be closed. I have updated the system call definitions with #395. I have used Ubuntu 16.04 sources, which add extra system calls compared to vanilla sources.

If vendors don't coordinate on system call number usage for system calls not merged in the upstream, this will cause some mis-reporting. However, #395 also updates the system call prototype scanning scripts. This makes it much easier to generate syscalls2 sources that fully cover another vendor's kernel.
